src/plugins/hetero/src/plugin.cpp

```diff
@@ -77,23 +81,105 @@ ov::hetero::Plugin::DeviceProperties ov::hetero::Plugin::get_properties_per_devi
     return device_properties;
 }
 
+void ov::hetero::Plugin::get_device_memory_map(const std::vector<std::string>& device_names,
+                                               std::map<std::string, size_t>& available_device_mem_map) const {
+    // TODO: add unified API to get device memory.
+    // There is no unified API to get device memory. So this feature get memory of specific device with specific method.
+    // Skip device which cannot get device memory size.
+    for (const auto& device_name : device_names) {
+        if (device_name.find("CPU") != std::string::npos) {
+            // Assuming the CPU has enough memory
+            available_device_mem_map["CPU"] = -1;
+        } else if (device_name.find("GPU") != std::string::npos) {
```

Comment:
ok, maybe we can keep as it is for now.
btw, what if gpu already have other workloads, is this device_total_mem_size can be used for estimate in this case? or should we leverage gpu statistics?
![image](https://github.com/openvinotoolkit/openvino/assets/32348500/16e7518f-5e76-4ead-8239-5fa9aaed0714)

Reply:
It will return statistics of GPU memory allocated by engine for each allocation type, we can combined it with ov::intel_gpu::device_total_mem_size to check if we can obtain more accurate device available memory. 
Based on our discussion, we will add this in the following optimization.